main.js

```diff
@@ -38,31 +38,31 @@ const removeClass = (el, oldClass) => {
 
     if (matchedObjects.length == 0) return;
 
-    matchedObjects.map((obj) => {
-      matchedEmojiTitles.push(obj.keywords[0]);
-    });
-
-    for (const emoji of matchedEmojiTitles) {
-      const emojiItem = document.querySelector(`[title="${emoji}"]`);
-
-      if (emojiItem) {
-        addClass(emojiItem.parentElement,'visible');
-      }
+    for (const matched of matchedObjects) {
+      setMatchedEmojiVisible(matched);
```

Comment:
(Referring to this but continuing our previous conversation)

Whether you want to do the two loops and then refactor to optimize is totally cool! But you also updated my filtering loop to include a function. So now I need to do it like this for mine which means merging this branch into `emoji-update` so I have your code.